src/rantanplan/core.py

```diff
@@ -376,18 +409,21 @@ def get_scansion(text, rhyme_analysis=False):
             seen_tokens.append(token)
     if len(seen_tokens) > 0:
         lines.append({"tokens": get_syllables(seen_tokens)})
-    lines = get_metre(lines)
+    for line in lines:
+        line.update(get_rhythmical_pattern(line, rhythm_format))
     if rhyme_analysis:
-        for rhyme in analyze_rhyme(lines):
-            for index, line in enumerate(lines):
-                line["structure"] = rhyme["name"]
-                line["rhyme"] = rhyme["rhyme"][index]
-                line["ending"] = rhyme["endings"][index]
-                line["ending_stress"] = rhyme["endings_stress"][index]
-                if line["ending_stress"] == 0:
-                    line["rhyme_type"] = ""
-                    line["rhyme_relaxation"] = None
-                else:
-                    line["rhyme_type"] = rhyme["rhyme_type"]
-                    line["rhyme_relaxation"] = rhyme["rhyme_relaxation"]
+        # Test
+        if analyze_rhyme(lines) is not None:
+            for rhyme in [analyze_rhyme(lines)]:
```

Comment:
No need for that comment I think. Plus, invoking twice `analyze_rhyme(lines)` is a bit wasteful.

Reply:
Maybe save it to a variable then?

Reply:
fixed doing just that, let me know if it's ok